projects/components/src/datagrid/interfaces/datagrid-column.interface.ts

```diff
@@ -40,6 +40,106 @@ export enum GridColumnSortDirection {
  */
 export type FunctionRenderer<T> = (record: T) => string;
 
+/**
+ * A type of button who's displayability does not depend on the selected entity.
+ */
+export interface GlobalButton {
+    /**
+     * The translated text of the button.
+     */
+    label: string;
+    /**
+     * The function that is called when the button is pressed.
+     */
+    handler: () => void;
+    /**
+     * The function that is called to determine if the button should be displayed.
+     */
+    shouldDisplay: () => boolean;
+    /**
+     * The css class the button should have.
+     */
+    class?: string;
+}
+
+/**
+ * A type of button who's displayability dependends on the selected entity.
+ */
+export interface ContextualButton<R> {
+    /**
+     * The translated text of the button.
+     */
+    label: string;
+    /**
+     * The function that is called when the button is pressed.
+     *
+     * @param entity the currently selected entities.
+     */
+    handler: (entity: R[]) => void;
+    /**
+     * The function that is called to determine if the button should be displayed.
+     *
+     * @param entity the currently selected entities.
+     */
+    shouldDisplay: (rec: R[]) => boolean;
```

Comment:
nit; I feel this would be better if it's named as isVisible or isDisabled

Reply:
Related to Juan's above comment about talking with Martin. Will update this when that conversation is resolved

Reply:
changes to isActive in the most recent commit